File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Python/s685734377_comments_removed.py
class FastScanner:
    def __init__(self, stream):
        import sys
        if stream == sys.stdin:
            self.stream = sys.stdin
        else:
            self.stream = open(stream, 'r')
        self.tokens = []
        self.position =  kop = 0

    def next(self):
        while self.position >= len(self.tokens):
            line = self.stream.readline()
            if not line:
                return None
            self.tokens = line.split()
            self.position = 0
        res = self.tokens[self.position]
        self.position += 1
        return res

    def nextInt(self):
        return int(self.next())

class TaskB:
    def solve(self, testNumber, in_, out):
        n = in_.nextInt()
        a = []
        for _ in range(n):
            s = in_.next()
            a.append(s)
        P = 37
        Q = 47
        powP = [1] * (n + 1)
        powQ = [1] * (n + 1)
        for i in range(1, n + 1):
            powP[i] = powP[i - 1] * P
            powQ[i] = powQ[i - 1] * Q
        hashRow = [[0] * n for _ in range(n)]
        hashCol = [[0] * n for _ in range(n)]
        for i in range(n):
            for j in range(n):
                for k in range(n):
                    exponent = (j - k) % n
                    term = ord(a[i][j]) * powP[exponent]
                    hashRow[i][k] += term
                    exponent_col = (i - k) % n
                    term_col = ord(a[i][j]) * powP[exponent_col]
                    hashCol[j][k] += term_col
        ans = 0
        for A in range(n):
            for B in range(n):
                ok = True
                for i in range(n):
                    row_idx = (i + A) % n
                    col_idx = (i + B) % n
                    if hashRow[row_idx][B] != hashCol[col_idx][A]:
                        ok = False
                        break
                if ok:
                    ans += 1
        out.write(f"{ans}\n")
        out.flush()

def main():
    import sys
    in_stream = sys.stdin
    out = sys.stdout
    scanner = FastScanner(in_stream)
    solver = TaskB()
    solver.solve(1, scanner, out)
    out.close()

if __name__ == "__main__":
    main()  # End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s685734377_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

type FastScanner struct {
	sc *bufio.Scanner
}

func NewFastScanner(r io.Reader) *FastScanner {
	sc := bufio.NewScanner(r)
	sc.Split(bufio.ScanWords)
	return &FastScanner{sc}
}

func (fs *FastScanner) Next() string {
	fs.sc.Scan()
	return fs.sc.Text()
}

func (fs *FastScanner) NextInt() int {
	n, _ := strconv.Atoi(fs.Next())
	return n
}

type TaskB struct{}

func (t *TaskB) Solve(testNumber int, in *FastScanner, out *bufio.Writer) {
	n := in.NextInt()
	a := make([][]byte, n)
	for i := 0; i < n; i++ {
		line := in.Next()
		a[i] = []byte(line)
	}

	const P = 37
	const Q = 47
	powP := make([]int64, n+1)
	powQ := make([]int64, n+1)
	powP[0] = 1
	powQ[0] = 1
	for i := 1; i <= n; i++ {
		powP[i] = powP[i-1] * int64(P)
		powQ[i] = powQ[i-1] * int64(Q)
	}

	hashRow := make([][]int64, n)
	hashCol := make([][]int64, n)
	for i := range hashRow {
		hashRow[i] = make([]int64, n)
		hashCol[i] = make([]int64, n)
	}

	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			for k := 0; k < n; k++ {
				jMinusK := (j - k + n) % n
				hashRow[i][k] += int64(a[i][j]) * powP[jMinusK]
				iMinusK := (i - k + n) % n
				hashCol[j][k] += int64(a[i][j]) * powP[iMinusK]
			}
		}
	}

	ans := 0
	for A := 0; A < n; A++ {
		for B := 0; B < n; B++ {
			ok := true
			for i := 0; i < n; i++ {
				rowI := (i + A) % n
				colI := (i + B) % n
				hr := hashRow[rowI][B]
				hc := hashCol[colI][A]
				if hr != hc {
					ok = false
					break
				}
			}
			if ok {
				ans++
			}
		}
	}
	fmt.Fprintf(out, "%d\n", ans)
}

func main() {
	in := NewFastScanner(os.Stdin)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()
	task := new(TaskB)
	task.Solve(1, in, out)
}

// End of Code
